aux_data.go

```diff
@@ -33,15 +40,18 @@ func auxDepositHasher() Sha256 {
 	return h
 }
 
-// ComputeAuxDataV0 Compute v0 AuxData given a ReferrerId
+// ComputeAuxData Compute the AuxData.
 //
 // This is defined as
 //
-//	taggedHash( Version0 || Nonce || ReferrerId )
+//	taggedHash( Version || Nonce || ReferrerId )
 //
-// where 'taggedHash' is a sha256 instance as returned by 'auxDepositHasher()',
-// 'Version0' is the byte 0x00, nonce, and 'ReferrerId' is an arbitrary 16 bytes array.
-func ComputeAuxDataV0(nonce uint32, referrerId []byte) ([]byte, error) {
+// where:
+// - 'taggedHash' is a sha256 instance as returned by 'auxDepositHasher()'
+// - 'Version' is a byte useful if different deposit versions should refer to different deposit addresses
+// - 'nonce' allows to generate different deposit addresses given same inputs
+// - 'ReferrerId' is an arbitrary 16 bytes array for application usage
+func ComputeAuxData(nonce uint32, referrerId []byte, version DepositAuxVersion) ([]byte, error) {
```

Comment:
let's do not brake compatibility with existing method and use different methods for each version. They could have different params in future.

Reply:
I kept the existing method, only added one more to be generic.

Reply:
Will add the V1 as well.

Reply:
I see, nit to make it private
```suggestion
func computeAuxData(nonce uint32, referrerId []byte, version DepositAuxVersion) ([]byte, error) {
```

Reply:
Fixed